stl/inc/flat_map

```diff
@@ -148,84 +151,88 @@ public:
+            return _Old;
         }
 
-        bool operator==(const type& _Right) const {
+        bool operator==(const _Iterator& _Right) const {
             return _Key_it == _Right._Key_it;
         }
 
-        auto operator<=>(const type& _Right) const {
+        auto operator<=>(const _Iterator& _Right) const {
             return _Key_it <=> _Right._Key_it;
         }
 
-        type& operator--() {
+        _Iterator& operator--() {
             --_Key_it;
             --_Mapped_it;
             return *this;
         }
 
-        type operator--(int) {
-            type _Tmp = *this;
+        _Iterator operator--(int) {
+            auto _Old = *this;
             --*this;
-            return _Tmp;
+            return _Old;
         }
 
-        type& operator+=(difference_type _Off) {
+        _Iterator& operator+=(const difference_type _Off) {
             _Key_it += _Off;
             _Mapped_it += _Off;
             return *this;
         }
 
-        type& operator-=(difference_type _Off) {
+        _Iterator& operator-=(const difference_type _Off) {
             _Key_it -= _Off;
             _Mapped_it -= _Off;
             return *this;
         }
 
-        type operator+(difference_type _Off) const {
-            type _Tmp = *this;
-            return _Tmp += _Off;
+        _Iterator operator+(const difference_type _Off) const {
+            auto _Old = *this;
+            _Old += _Off;
+            return _Old;
         }
 
-        type operator-(difference_type _Off) const {
-            type _Tmp = *this;
-            return _Tmp -= _Off;
+        _Iterator operator-(const difference_type _Off) const {
+            auto _Old = *this;
+            _Old -= _Off;
+            return _Old;
         }
 
-        reference operator[](difference_type _Off) const {
+        reference operator[](const difference_type _Off) const {
             return *(*this + _Off);
         }
 
-        difference_type operator-(const type& _Right) const {
+        difference_type operator-(const _Iterator& _Right) const {
             return _Key_it - _Right._Key_it;
         }
 
-        friend type operator+(difference_type _Off, const type& _Right) {
+        friend _Iterator operator+(const difference_type _Off, const _Iterator& _Right) {
             return _Right + _Off;
         }
 
-        operator typename _Flat_map_iterator_Impl<_KeyContainer, _MappedContainer, true>::type() const
-            requires (!_IsConst)
+        operator typename _Paring_iterator_provider<_KeyContainer, _MappedContainer,
+            _Paring_iterator_kind::_Const>::_Iterator() const
+            requires (_Kind == _Paring_iterator_kind::_Mutable)
         {
-            return typename _Flat_map_iterator_Impl<_KeyContainer, _MappedContainer, true>::type{_Key_it, _Mapped_it};
+            return typename _Paring_iterator_provider<_KeyContainer, _MappedContainer,
+                _Paring_iterator_kind::_Const>::_Iterator{_Key_it, _Mapped_it};

```

Comment:
`typename _Paring_iterator_provider<_KeyContainer, _MappedContainer, _Paring_iterator_kind::_Const>::_Iterator` is horribly long and repeated twice. Can we extract a private typedef here?